### ISSUE TITLE
Embedded multipleinput html attributes

### ISSUE THREAD

**Issue by bokodi-dev** [unclead/yii2-multiple-input#92]
The embedded multipleinput elements share the same id and name html attributes.

e.g.
```
MultipleInput::widget([
    'model' => $model,
    'attribute' => 'options',
    'columns' => [
        [
            'name' => 'name',
            'title' => 'name',
        ],
        [
            'name' => 'values',
            'type'  => MultipleInput::class,
            'options' => [
                'columns' => [
                    ['name' => 'value', 'title' => 'value'],
                    ['name' => 'amount', 'title' => 'amount'],
                    [
                        'name' => 'type',
                        'title' => 'type',
                        'type' => \kartik\widgets\Select2::className(),
                        'options' => [
                            'data' => ['data1', 'data2']
                        ]
                    ],
                ]
            ]
        ]
    ],
]);
```

When there are more `options` and those options also have more `values`, the name attributes look like this:
```
Option[0][name]
Option[values][0][value]
Option[values][0][amount]
Option[values][0][type]
Option[values][1][value]
Option[values][1][amount]
Option[values][1][type]
Option[1][name]
Option[values][0][value]
Option[values][0][amount]
Option[values][0][type]
```

As you can see, there are duplicated names. The ids are also duplicated. The name values in this example should look like this:
```
Option[0][name]
Value[0][0][value]
Value[0][0][amount]
Value[0][0][type]
Value[0][1][value]
Value[0][1][amount]
Value[0][1][type]
Option[1][name]
Value[1][0][value]
Value[1][0][amount]
Value[1][0][type]
```
This issue also causes kartik select2 widget elements to break (due to the duplicated ids).

**Comment by unclead** [comment]
Do you use last version 1.3.1?

**Comment by bokodi-dev** [comment]
Yes, I have the latest version (1.3.1)

**Comment by unclead** [comment]
Hmmm i cannot reproduce the bug.

Here is a POST request from my test:

```
ExampleModel[options][0][name]:test
ExampleModel[options][0][values][0][value]:
ExampleModel[options][0][values][0][amount]:
ExampleModel[options][0][values][0][type]:0
ExampleModel[options][0][values][1][value]:
ExampleModel[options][0][values][1][amount]:
ExampleModel[options][0][values][1][type]:0
ExampleModel[options][0][values][2][value]:
ExampleModel[options][0][values][2][amount]:
ExampleModel[options][0][values][2][type]:0
ExampleModel[options][1][name]:
ExampleModel[options][1][values][0][value]:
ExampleModel[options][1][values][0][amount]:
ExampleModel[options][1][values][0][type]:0
```

as you can seethere are no duplicates

**Comment by unclead** [comment]
this is my view file

```
<?php

use yii\bootstrap\ActiveForm;
use unclead\widgets\MultipleInput;
use unclead\widgets\examples\models\ExampleModel;
use yii\helpers\Html;
use unclead\widgets\MultipleInputColumn;


/* @var $this \yii\web\View */
/* @var $model ExampleModel */

$commonAttributeOptions = [
    'enableAjaxValidation'   => false,
    'enableClientValidation' => false,
    'validateOnChange'       => false,
    'validateOnSubmit'       => true,
    'validateOnBlur'         => false,
];

$enableActiveForm = false;
?>

<?php

if ($enableActiveForm) {
    $form = ActiveForm::begin([
        'enableAjaxValidation'      => true,
        'enableClientValidation'    => false,
        'validateOnChange'          => false,
        'validateOnSubmit'          => true,
        'validateOnBlur'            => false,
    ]);
} else {
    echo Html::beginForm();
}

?>

<?= MultipleInput::widget([
    'model' => $model,
    'attribute' => 'options',
    'columns' => [
        [
            'name' => 'name',
            'title' => 'name',
        ],
        [
            'name' => 'values',
            'type'  => MultipleInput::class,
            'options' => [
                'columns' => [
                    ['name' => 'value', 'title' => 'value'],
                    ['name' => 'amount', 'title' => 'amount'],
                    [
                        'name' => 'type',
                        'title' => 'type',
                        'type' => MultipleInputColumn::TYPE_DROPDOWN,
                        'items' => [
                            'data' => ['data1', 'data2']
                        ]
                    ],
                ]
            ]
        ]
    ],
]); ?>

<?= Html::submitButton('Update', ['class' => 'btn btn-success']);?>
<?php
if ($enableActiveForm) {
    ActiveForm::end();
} else {
    echo Html::endForm();
}
?>
```

**Comment by bokodi-dev** [comment]
I tested with cleared assets.

Here is my actual code:
```
<?= MultipleInput::widget([
    'model' => $model,
    'attribute' => 'shopOptions',
    'columns' => [
        [
            'name' => 'name',
            'title' => 'name',
        ],
        [
            'name' => 'shopOptionValues',
            'type'  => MultipleInput::class,
            'options' => [
                'columns' => [
                    ['name' => 'value', 'title' => 'value'],
                    ['name' => 'price_modify', 'title' => 'price_modify'],
                    [
                        'name' => 'price_modify_type',
                        'title' => 'price_modify_type',
                        'type' => \kartik\widgets\Select2::className(),
                        'options' => [
                            'data' => [ShopOptionValue::TYPE_SUBTRACTION => ucfirst(ShopOptionValue::TYPE_SUBTRACTION), ShopOptionValue::TYPE_ADDITION => ucfirst(ShopOptionValue::TYPE_ADDITION)]
                        ],
                    ],
                    ['name' => 'stock_amount', 'title' => 'stock_amount'],
                ]
            ]
        ]
    ],
]); ?>
```

In [this screenshot](http://prnt.sc/byxisa) you can see the duplicated ids, and the missing select2 widget.

**Comment by unclead** [comment]
Can you make a screenshot with ID of other inputs, e.g. `name`.
Also it will help me if you show me full code, not only the code of a view file.

I took your code and tried to reproduce the bug on clear instance and it works for me
here is a code: https://gist.github.com/unclead/a3202d6e83e52ec85eb24ea2149069ed

What am i do wrong?

**Comment by bokodi-dev** [comment]
The ids on the top level are not duplicated: [screen](http://prnt.sc/bz0ssf).
But the embedded ones are still wrong: [screen](http://prnt.sc/bz0tcx).

```
class ShopOptionSet extends ActiveRecord {
    public static function tableName() {
        return 'shop_option_set';
    }

    public function getShopOptions() {
        return $this->hasMany(ShopOption::className(), ['option_set_id' => 'option_set_id']);
    }
}

class ShopOption extends ActiveRecord {
    public static function tableName() {
        return 'shop_option';
    }

    public function getOptionSet() {
        return $this->hasOne(ShopOptionSet::className(), ['option_set_id' => 'option_set_id']);
    }

    public function getShopOptionValues() {
        return $this->hasMany(ShopOptionValue::className(), ['option_id' => 'option_id']);
    }
}

class ShopOptionValue extends ActiveRecord {
    public static function tableName() {
        return 'shop_option_value';
    }

    public function getOption() {
        return $this->hasOne(ShopOption::className(), ['option_id' => 'option_id']);
    }
}
```

**Comment by unclead** [comment]
Hmmm i've never tried to use widget with AR relations. I have all necessary info now and need time to dig deeper.

**Comment by unclead** [comment]
fixed 8f370af9ae390025a6300e1521b4b384055b83d6
